scripts/variantstore/wdl/GvsCreateVATAnnotations.wdl

```diff
@@ -111,64 +115,81 @@ task ExtractAnAcAfFromVCF {
+        bcftools view --threads 4 -i 'N_ALT>50 || REF~"N"' -O u original.bcf | bcftools query -f '%CHROM\t%POS\t%ID\t%REF\t%ALT\n' > track_dropped.tsv
 
         wc -l track_dropped.tsv
 
-        ## filter out sites with too many alt alleles
-        bcftools view -e 'N_ALT>50 || REF~"N"' --no-update ~{local_input_vcf} -Ou | \
+        echo_date "VAT: filter out sites with too many alt alleles"
+        bcftools view --threads 4 -e 'N_ALT>50 || REF~"N"' --no-update original.bcf -O u | \
         ## filter out the non-passing sites
-        bcftools view  -f 'PASS,.' --no-update -Oz -o filtered.vcf.gz
-        ## normalize, left align and split multi allelic sites to new lines, remove duplicate lines
-        bcftools norm -m- --check-ref w -f Homo_sapiens_assembly38.fasta filtered.vcf.gz -Oz -o normalized.vcf.gz
-        rm ~{local_input_vcf}
-        ## filter out spanning deletions and variants with an AC of 0
-        bcftools view -e 'ALT[0]="*" || AC=0' --no-update normalized.vcf.gz -Ou | \
+        bcftools view --threads 4 -f 'PASS,.' --no-update -O b -o filtered.bcf
+
+        echo_date "VAT: normalize, left align and split multi allelic sites to new lines, remove duplicate lines"
+        bcftools norm --threads 4 -m- --check-ref w -f Homo_sapiens_assembly38.fasta filtered.bcf -O b -o normalized.bcf
+        rm filtered.bcf
+
+        echo_date "VAT: filter out spanning deletions and variants with an AC of 0, respect the FT flag"
+        bcftools view --threads 4 -e 'ALT[0]="*" || AC=0' --no-update normalized.bcf -O u | \
         ## ensure that we respect the FT tag
-        bcftools filter -i "FORMAT/FT='PASS,.'" --set-GTs . -Oz -o ~{normalized_vcf}
+        bcftools filter --threads 4 -i "FORMAT/FT='PASS,.'" --set-GTs . -O b -o normalized.filtered.bcf
 
         ## clean up unneeded file
-        rm normalized.vcf.gz
+        rm normalized.bcf
+
+        echo_date "VAT: detecting and removing duplicate rows"
 
         ## During normalization, sometimes duplicate variants appear but with different calculations. This seems to be a bug in bcftools. For now we are dropping all duplicate variants
         ## to locate the duplicates, we first make a file of just the first 5 columns
-        bcftools query ~{normalized_vcf} -f '%CHROM\t%POS\t%ID\t%REF\t%ALT\n' > check_duplicates.tsv
-        ## check it for duplicates and put them in a new file
-        sort check_duplicates.tsv | uniq -d | cut -f1,2,3,4,5  > duplicates.tsv
-        rm check_duplicates.tsv ## clean up
-        ## remove those rows (that match up to the first 5 cols)
-        zgrep -v -wFf duplicates.tsv ~{normalized_vcf} > deduplicated.vcf.gz
-        rm ~{normalized_vcf} ## clean up
+        bcftools query normalized.filtered.bcf -f '%CHROM\t%POS\t%ID\t%REF\t%ALT\n' | sort check_duplicates.tsv | uniq -d | cut -f1,2,3,4,5 > duplicates.tsv
```

Comment:
I think this can be simplified to:
grep -v '^#' normalized.filtered.bcf | cut -f 1-5 | sort | uniq -d  > duplicates.tsv

Reply:
and actually we dont need ID either so we could do it with f1,2,4,5